mila_datamodules/cli/prepare_torchvision.py

```diff
@@ -45,56 +58,78 @@ def __call__(
 
         logger.debug(
-            f"Using dataset constructor: {self.dataset_type} with args {constructor_args}, and "
-            f"kwargs {constructor_kwargs}"
+            f"Using dataset constructor: {self.dataset_type} with args {dataset_args}, and "
+            f"kwargs {dataset_kwargs}"
         )
-        dataset_instance = self.dataset_type(str(root), *constructor_args, **constructor_kwargs)
+        dataset_instance = self.dataset_type(str(root), *dataset_args, **dataset_kwargs)
         if is_local_main():
             print(dataset_instance)
         return str(root)
 
 
-class SymlinkArchives(PrepareVisionDataset[VD, P]):
-    """Creates symlinks to the archives in the `root` directory."""
+class SymlinkDatasetFiles(AbstractPrepareVisionDataset[VD, P]):
+    """Creates symlinks to the datasets' files in the `root` directory."""
 
     def __init__(
         self,
-        dataset_type: Callable[Concatenate[str, P], VD],
-        relative_paths_to_archives: dict[str, str | Path],
+        source: str | Path = None,
+        files: dict[str, str | Path] = None
     ):
         """
-
         Parameters
         ----------
 
-        - dataset_type:
-            Callable that returns an instance of a `torchvision.datasets.VisionDataset`.
-
-        - relative_paths_to_archives:
-            A mapping from a relative path where the symlink to the archive should be created
+        - source:
+            Source directory to clone the drectories and files hieararchy. If
+            only source is defined, the entire tree will be cloned.
+        - files:
+            A mapping from a path to where the symlink to the archive should be created
             (relative to the 'root' directory) to the actual path to the archive on the cluster.
         """
-        self.dataset_type = dataset_type
-        self.relative_paths_to_archives = {
-            k: Path(v) for k, v in relative_paths_to_archives.items()
-        }
+        if source is not None:
+            source = Path(source).expanduser()
+
+        if source is not None and files is not None:
+            self.relative_paths_to_files = {
+                k: Path(v).relative_to(source) for k, v in files.items()
+            }
+        elif source is not None:
+            self.relative_paths_to_files = {
+                _f.relative_to(source): _f
+                for _f in SymlinkDatasetFiles._recursive_list_files(Path(source))
+            }
+        else:
+            self.relative_paths_to_files = {
+                k: Path(v) for k, v in files.items()
+            }
+
+    @staticmethod
+    def _recursive_list_files(root:Path):
+        for entry in root.iterdir():
+            if entry.name.startswith("."):
+                continue
+            if entry.is_file():
+                yield entry
+            if entry.is_dir():
+                for _f in SymlinkDatasetFiles._recursive_list_files(entry):
+                    yield _f
```

Comment:
Nitpick
```suggestion
                yield from SymlinkDatasetFiles._recursive_list_files(entry)
```

Reply:
So much better. I was wondering why it was so ugly to yield a generator in Python. Turns out it was just me